state/state.go

```diff
@@ -303,7 +288,10 @@ func MakeServer(config config.ServerConfig, testRun bool) *Server {
 		ActivityStateMutex:    &sync.Mutex{},
 		CollectionStatusMutex: &sync.Mutex{},
 		LogTimezoneMutex:      &sync.Mutex{},
+		SnapshotStream:        make(chan []byte),
 	}
+	server.Pause = pganalyze_collector.ServerMessage_Pause{Pause: false}
+	server.Grant.Config.Features = &pganalyze_collector.ServerMessage_Features{}
```

Comment:
This was needed to avoid a nil pointer error, since the protobuf library doesn't have an option to nest structs without using pointers https://github.com/golang/protobuf/issues/1225

Reply:
Hmm, apparently the `integration_test` command [failed on CI](https://github.com/pganalyze/collector/actions/runs/10406859790/job/28820868142) for the same reason. Any idea how to fix this?

Reply:
It's failing because that integration test is doing the `--dry-run --force-state-update` (you can repro locally, actually even without `--force-state-update`).

With that test, it runs the full snapshot dry run test first, then move on to the activity snapshot dry run test. After the successful full snapshot test, `server.Grant` is updated with `newGrant, err = grant.GetDefaultGrant(ctx, server, globalCollectionOpts, logger)`, so it will become something like `state.Grant{}`. As you worked around here, you likely need to do similar operation for other places too to make sure that `Grant.Config.Features` won't be nil.

Is it possible to implement this part without introducing this change? Feels like that might be simpler. Or, keep the struct in a previous way and update it more manually (instead of doing `server.Grant.Config = *message.GetConfig()`, update one by one, for example).

Reply:
Thanks, that's been fixed in 98b6cdd10b86a0cf919b6bdba092f0013f9e256a.

Yeah this nil pointer issue is unfortunate. I changed the `state.Grant` to contain the protobuf struct directly to avoid the two versions of the struct from getting out of sync. At the time I didn't realize this would be an issue.